libs/barista-components/stacked-bar-chart/src/stacked-bar-chart.ts

```diff
@@ -0,0 +1,471 @@
+      this._series = value;
+      this._legends = getLegends(value, this._theme);
+      this._filledSeries = fillSeries(value, this._legends);
+
+      this._canShowValue = this._series.length === 1;
+
+      this._render();
+    }
+  }
+  private _series: DtStackedBarChartSeries[];
+  /** Series with filled nodes */
+  private _filledSeries: DtStackedBarChartFilledSeries[];
+
+  /** Allow selections to be made on chart */
+  @Input()
+  get selectable(): boolean {
+    return this._selectable;
+  }
+  set selectable(value: boolean) {
+    if (value !== this._selectable) {
+      this._toggleSelect();
+      this._selectable = value;
+    }
+  }
+  _selectable: boolean = false;
+
+  /** Current selection [series, node] */
+  @Input()
+  get selected(): [DtStackedBarChartSeries, DtStackedBarChartNode] | [] {
+    return this._selected;
+  }
+  set selected([series, node]:
+    | [DtStackedBarChartSeries, DtStackedBarChartNode]
+    | []) {
+    // if selected node is different than current
+    if (this._selected[1] !== node) {
+      this._toggleSelect(series, node);
+    }
+  }
+  private _selected: [DtStackedBarChartSeries, DtStackedBarChartNode] | [] = [];
+
+  @Output() selectedChange: EventEmitter<
+    [DtStackedBarChartSeries, DtStackedBarChartNode] | []
+  > = new EventEmitter();
+
+  /** Max value in the chart */
+  @Input()
+  get max(): number | undefined {
+    return this._max !== undefined ? this._max : getTotalMaxValue(this.series);
+  }
+  set max(value: number | undefined) {
+    if (value !== this._max) {
+      this._max = value;
+      this._render();
+    }
+  }
+  private _max: number | undefined;
+
+  /** Whether each bar should be filled completely or should take into account their siblings and max  */
+  @Input()
+  get fillMode(): DtStackedBarChartFillMode {
+    return this._fillMode;
+  }
+  set fillMode(value: DtStackedBarChartFillMode) {
+    if (value !== this._fillMode) {
+      this._fillMode = value;
+      this._render();
+    }
+  }
+  private _fillMode: DtStackedBarChartFillMode = 'relative';
+
+  /** Sets the display mode for the stacked-bar-chart values in legend to either 'none' 'percent' or 'absolute'. Only valid for single track chart.  */
+  @Input() valueDisplayMode: DtStackedBarChartValueDisplayMode = 'none';
+  /** @internal Will be true if display mode is not 'none' and only has one series */
+  _canShowValue: boolean;
+
+  /** Array of legends that can be used to toggle bar nodes. Useful when the legend used is outside this component */
+  @Input()
+  get legends(): DtStackedBarChartLegend[] {
+    return this._legends;
+  }
+  set legends(values: DtStackedBarChartLegend[]) {
+    if (values !== undefined && values !== this._legends) {
+      this._legends = values;
+      updateNodesWithLegend(this._filledSeries, this._legends);
+
+      this._render();
+    }
+  }
+  _legends: DtStackedBarChartLegend[];
+
+  /** Visibility of the legend */
+  @Input() visibleLegend: boolean = true;
```

Comment:
@tomheller I used `visibleWhatever` instead of `showWhatever` because show is an action and it's a bit weird to be using that for a variable. But if we have a policy of naming it as you suggest I'll do the changes. What should I do?

Reply:
Your reasoning makes a lot of sense. Let's stick with `visibleWhatever` 👍